latent_programmer/spec_decomposition/decomposition_models.py

```diff
@@ -246,12 +280,18 @@ def decode(self,
         else:
           decoder_relative_position = None
 
-      encoder_decoder_mask = nn.make_attention_mask(
-          programs > 0, flat_encoded_padding_mask, dtype=cfg.dtype)
+        if self.config.separator_special_attention:
```

Comment:
let's check that self.config.separator_token is not -1, raising ValueError if it is, so we are notified if we forgot to set separator_token correctly